ts/src/cryptocom.ts

```diff
@@ -363,206 +363,167 @@ export default class cryptocom extends Exchange {
+            const marginSellEnabled = this.safeValue (market, 'margin_sell_enabled');
             let expiry = this.safeInteger (market, 'expiry_timestamp_ms');
             if (expiry === 0) {
                 expiry = undefined;
             }
-            let type = 'swap';
-            if (future) {
+            let symbol = base + '/' + quote;
+            let type = undefined;
+            let contract = undefined;
+            if (inst_type === 'CCY_PAIR') {
+                type = 'spot';
+                contract = false;
+            } else if (inst_type === 'PERPETUAL_SWAP') {
+                type = 'swap';
+                symbol = symbol + ':' + quote;
+                contract = true;
+            } else if (inst_type === 'FUTURE') {
                 type = 'future';
-                symbol = symbol + '-' + this.yymmdd (expiry);
+                symbol = symbol + ':' + quote + '-' + this.yymmdd (expiry);
+                contract = true;
+            } else if (inst_type === 'WARRANT') {
+                type = 'option';
+                const symbolOptionType = (optionType === 'call') ? 'C' : 'P';
+                symbol = symbol + ':' + quote + '-' + this.yymmdd (expiry) + '-' + strike + '-' + symbolOptionType;
+                contract = true;
             }
-            const contractSize = this.safeNumber (market, 'contract_size');
             result.push ({
                 'id': this.safeString (market, 'symbol'),
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
-                'settle': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'settleId': quoteId,
+                'settleId': settleId,
                 'type': type,
-                'spot': false,
-                'margin': false,
+                'spot': spot,
+                'margin': ((marginBuyEnabled) || (marginSellEnabled)),
                 'swap': swap,
                 'future': future,
-                'option': false,
+                'option': option,
                 'active': this.safeValue (market, 'tradable'),
-                'contract': true,
-                'linear': true,
-                'inverse': false,
-                'contractSize': contractSize,
+                'contract': contract,
+                'linear': (contract) ? true : undefined,
+                'inverse': (contract) ? false : undefined,
+                'contractSize': this.safeNumber (market, 'contract_size'),
                 'expiry': expiry,
                 'expiryDatetime': this.iso8601 (expiry),
-                'strike': undefined,
-                'optionType': undefined,
+                'strike': this.parseNumber (strike),
+                'optionType': optionType,
                 'precision': {
                     'price': this.parseNumber (this.parsePrecision (this.safeString (market, 'quote_decimals'))),
```

Comment:
@Dan-krm  can't we use these values directly?
![image](https://github.com/ccxt/ccxt/assets/43336371/e0f0f9f3-be29-4fb8-8bab-f3b4c3dacc9a)